### PR TITLE
[docs] Translate mysql cdc doc to Chinese

### PR DESCRIPTION
**Pull request by rookiegao**
This PR Main solutions #1391

There may be inaccuracies or omissions in the translation of the first edition.

### CODE REVIEW

File: docs/content/connectors/mysql-cdc(ZH).md [hunk 1]
```diff
@@ -0,0 +1,884 @@
...
+### 为每个 reader 设置不同的服务器 ID
+
+每个用于读取 binlog 的 MySQL 数据库客户端都应该有一个唯一的 id，称为服务器 id。 MySQL 服务器将使用此 id 来维护网络连接和 binlog 位置。 因此，如果不同的作业共享相同的服务器 id， 则可能导致从错误的 binlog 位置读取数据。
+因此，建议通过为每个 Reader 设置不同的服务器 id  [SQL Hints](https://ci.apache.org/projects/flink/flink-docs-release-1.11/dev/table/sql/hints.html),
+假设 Source 并行度为 4, 我们可以使用 `SELECT * FROM source_table /*+ OPTIONS('server-id'='5401-5404') */ ;` 来为 4 个 Source Readers 中的每一个分配唯一的服务器 id。
+
+
+### 设置 MySQL 会话超时时间
+
+当为大型数据库创建初始一致快照时，您建立的连接可能会在读取表时超时。您可以通过在MySQL配置文件中配置 interactive_timeout 和 wait_timeout 来防止这种行为。
```
Comment (leonardBang) [inline docs/content/connectors/mysql-cdc(ZH).md:74]:
```suggestion
当为大型数据库创建初始一致快照时，您建立的连接可能会在读取表时碰到超时问题。您可以通过在 MySQL 侧配置 interactive_timeout 和 wait_timeout 来缓解此类问题。
```

File: docs/content/connectors/mysql-cdc(ZH).md [hunk 2]
```diff
@@ -0,0 +1,884 @@
...
+- `wait_timeout`: 服务器在关闭非交互连接之前等待活动的秒数。 更多信息请参考 [MySQL documentations](https://dev.mysql.com/doc/refman/8.0/en/server-system-variables.html#sysvar_wait_timeout).
+
+
+如何创建 MySQL CDC 表
+----------------
+
+MySQL CDC 表可以定义如下：
+
+```sql
+-- 每 3000 毫秒做一次 checkpoint
```
Comment (leonardBang) [inline docs/content/connectors/mysql-cdc(ZH).md:85]:
```suggestion
-- 每 3 秒做一次 checkpoint，用于测试，生产配置建议5到10分钟                      
```

File: docs/content/connectors/mysql-cdc(ZH).md [hunk 3]
```diff
@@ -0,0 +1,884 @@
...
+     ) WITH (
+     'connector' = 'mysql-cdc',
+     'hostname' = 'localhost',
+     'port' = '3306',
+     'username' = 'root',
+     'password' = '123456',
+     'database-name' = 'mydb',
+     'table-name' = 'orders');
+  
+-- 从订单表读取快照和 binlog
```
Comment (leonardBang) [inline docs/content/connectors/mysql-cdc(ZH).md:106]:
```suggestion
-- 从订单表读取全量数据(快照)和增量数据(binlog)
```

File: docs/content/connectors/mysql-cdc(ZH).md [hunk 4]
```diff
@@ -0,0 +1,884 @@
...
+    <tr>
+          <td>scan.incremental.snapshot.enabled</td>
+          <td>optional</td>
+          <td style="word-wrap: break-word;">true</td>
+          <td>Boolean</td>
+          <td>增量快照是一种读取表快照的新机制。 与旧的快照机制相比，
+              增量快照有许多优点，包括：
+              （1） 在快照读取期间，Source 可以是并行的，
+              （2） Source 可以在快照读取期间在区块粒度中执行 checkpoint，
+              （3） 在快照读取之前，Source 不需要获取全局读取锁（使用读取锁刷新表）。
```
Comment (leonardBang) [inline docs/content/connectors/mysql-cdc(ZH).md:193]:
```suggestion
              （1） 在快照读取期间，Source 支持并发读取，
              （2）在快照读取期间， Source 支持进行 chunk 粒度的checkpoint，
              （3）在快照读取之前，Source 不需要数据库锁权限。
```

File: docs/content/connectors/mysql-cdc(ZH).md [hunk 5]
```diff
@@ -0,0 +1,884 @@
...
+          <td>Integer</td>
+          <td>读取表快照时每次轮询的最大获取大小。</td>
+    </tr>
+    <tr>
+      <td>scan.startup.mode</td>
+      <td>optional</td>
+      <td style="word-wrap: break-word;">initial</td>
+      <td>String</td>
+      <td> MySQL CDC 消费者的可选启动模式， 有效枚举为 "initial"
+           and "latest-offset".
```
Comment (leonardBang) [inline docs/content/connectors/mysql-cdc(ZH).md:219]:
```suggestion
         合法的模式为 "initial" 和 "latest-offset"。
```

File: docs/content/connectors/mysql-cdc(ZH).md [hunk 6]
```diff
@@ -0,0 +1,884 @@
...
+          <td>读取表快照时每次轮询的最大获取大小。</td>
+    </tr>
+    <tr>
+      <td>scan.startup.mode</td>
+      <td>optional</td>
+      <td style="word-wrap: break-word;">initial</td>
+      <td>String</td>
+      <td> MySQL CDC 消费者的可选启动模式， 有效枚举为 "initial"
+           and "latest-offset". 
+           请查阅 <a href="#startup-reading-position">Startup Reading Position</a> 章节了解更多详细信息。</td>
```
Comment (leonardBang) [inline docs/content/connectors/mysql-cdc(ZH).md:220]: 这个链接点下能否访问

File: docs/content/connectors/mysql-cdc(ZH).md [hunk 7]
```diff
@@ -0,0 +1,884 @@
...
+    </tr> 
+    <tr>
+      <td>server-time-zone</td>
+      <td>optional</td>
+      <td style="word-wrap: break-word;">UTC</td>
+      <td>String</td>
+      <td>数据库服务器中的会话时区， 例如： "Asia/Shanghai". 
+          它控制 MYSQL 中的时间戳类型如何转换为字符串。
+          更多请参考 <a href="https://debezium.io/documentation/reference/1.5/connectors/mysql.html#mysql-temporal-types"> 这里</a>.</td>
```
Comment (leonardBang) [inline docs/content/connectors/mysql-cdc(ZH).md:229]:
```suggestion
          更多请参考 <a href="https://debezium.io/documentation/reference/1.6/connectors/mysql.html#mysql-temporal-types"> 这里</a>.</td>
```

File: docs/content/connectors/mysql-cdc(ZH).md [hunk 8]
```diff
@@ -0,0 +1,884 @@
...
+      <td>String</td>
+      <td>将 Debezium 的属性传递给 Debezium 嵌入式引擎，该引擎用于从 MySQL 服务器捕获数据更改。
+          For example: <code>'debezium.snapshot.mode' = 'never'</code>.
+          查看更多关于 <a href="https://debezium.io/documentation/reference/1.5/connectors/mysql.html#mysql-connector-properties"> Debezium 的  MySQL 连接器属性</a></td> 
+    </tr>
+    </tbody>
+</table>
+</div>
+
+可用元数据
```
Comment (leonardBang) [inline docs/content/connectors/mysql-cdc(ZH).md:288]: 支持的元数据

File: docs/content/connectors/mysql-cdc(ZH).md [hunk 9]
```diff
@@ -0,0 +1,884 @@
...
+----------------
+
+以下格式元数据可以在表定义中作为只读（虚拟）列公开。
```
Comment (leonardBang) [inline docs/content/connectors/mysql-cdc(ZH).md:291]:
```suggestion
下表中的元数据可以在 DDL 中作为只读（虚拟）meta 列声明。
```

File: docs/content/connectors/mysql-cdc(ZH).md [hunk 10]
```diff
@@ -0,0 +1,884 @@
...
+       <th class="text-left" style="width: 15%">Key</th>
+       <th class="text-left" style="width: 30%">DataType</th>
+       <th class="text-left" style="width: 55%">Description</th>
+     </tr>
+  </thead>
+  <tbody>
+    <tr>
+      <td>table_name</td>
+      <td>STRING NOT NULL</td>
+      <td>包含行的表的名称。</td>
```
Comment (leonardBang) [inline docs/content/connectors/mysql-cdc(ZH).md:305]: 当前记录所属的表名称。

File: docs/content/connectors/mysql-cdc(ZH).md [hunk 11]
```diff
@@ -0,0 +1,884 @@
...
+    </tr>
+    <tr>
+      <td>database_name</td>
+      <td>STRING NOT NULL</td>
+      <td>包含行的数据库的名称。</td>
```
Comment (leonardBang) [inline docs/content/connectors/mysql-cdc(ZH).md:310]: 当前记录所属的库名称。

File: docs/content/connectors/mysql-cdc(ZH).md [hunk 12]
```diff
@@ -0,0 +1,884 @@
...
+    </tr>
+    <tr>
+      <td>op_ts</td>
+      <td>TIMESTAMP_LTZ(3) NOT NULL</td>
+      <td>它表示在数据库中进行更改的时间。 <br>如果从表的快照而不是 binlog 读取记录，该值将始终为0。</td>
```
Comment (leonardBang) [inline docs/content/connectors/mysql-cdc(ZH).md:315]:
```suggestion
      <td>当前记录表在数据库中更新的时间。 <br>如果从表的快照而不是 binlog 读取记录，该值将始终为0。</td>
```

File: docs/content/connectors/mysql-cdc(ZH).md [hunk 13]
```diff
@@ -0,0 +1,884 @@
...
+    </tr>
+  </tbody>
+</table>
+
+扩展创建表示例演示了公开这些元数据字段的语法：
```
Comment (leonardBang) [inline docs/content/connectors/mysql-cdc(ZH).md:320]:
```suggestion
下述创建表示例展示元数据列的用法：
```

File: docs/content/connectors/mysql-cdc(ZH).md [hunk 14]
```diff
@@ -0,0 +1,884 @@
...
+    'hostname' = 'localhost',
+    'port' = '3306',
+    'username' = 'root',
+    'password' = '123456',
+    'database-name' = 'mydb',
+    'table-name' = 'orders'
+);
+```
+
+Features
```
Comment (leonardBang) [inline docs/content/connectors/mysql-cdc(ZH).md:344]: 支持的特性

File: docs/content/connectors/mysql-cdc(ZH).md [hunk 15]
```diff
@@ -0,0 +1,884 @@
...
+--------
+
+### Incremental Snapshot Reading
```
Comment (leonardBang) [inline docs/content/connectors/mysql-cdc(ZH).md:347]: 增量快照读取

File: docs/content/connectors/mysql-cdc(ZH).md [hunk 16]
```diff
@@ -0,0 +1,884 @@
...
+
+增量快照读取是一种读取表快照的新机制。与旧的快照机制相比，增量快照具有许多优点，包括：
+* （1）MySQL CDC Source 可以在快照读取期间并行
+* （2）MySQL CDC Source 可以在快照读取期间在区块粒度中执行检查点
+* （3）MySQL CDC Source 不需要在快照读取之前获取全局读取锁（使用读取锁刷新表）
```
Comment (leonardBang) [inline docs/content/connectors/mysql-cdc(ZH).md:352]: 同上

File: docs/content/connectors/mysql-cdc(ZH).md [hunk 17]
```diff
@@ -0,0 +1,884 @@
...
+
+如果希望 source 并行运行，则每个并行 reader 都应该具有唯一的服务器 id，因此“服务器 id”的范围必须类似于 “5400-6400”，
+且范围必须大于并行度。
+
+在增量快照读取过程中，MySQL CDC Source 首先通过表的主键拆分快照块（splits），
+然后 MySQL CDC Source 将区块分配给多个 reader 以读取快照区块的数据。
```
Comment (leonardBang) [inline docs/content/connectors/mysql-cdc(ZH).md:358]: 不用换行

File: docs/content/connectors/mysql-cdc(ZH).md [hunk 18]
```diff
@@ -0,0 +1,884 @@
...
+
+#### 控制作业并行度
```
Comment (leonardBang) [inline docs/content/connectors/mysql-cdc(ZH).md:360]: 支持并发读取

File: docs/content/connectors/mysql-cdc(ZH).md [hunk 19]
```diff
@@ -0,0 +1,884 @@
...
+
+增量快照读取提供了并行读取快照数据的能力。
+你可以通过设置作业并行度的方式来控制 Source 的并行度 `parallelism.default`. For example, in SQL CLI:
+
+```sql
+Flink SQL> SET 'parallelism.default' = 8;
+```
+
+#### Checkpoint
```
Comment (leonardBang) [inline docs/content/connectors/mysql-cdc(ZH).md:369]: 全量阶段支持 checkpoint

File: docs/content/connectors/mysql-cdc(ZH).md [hunk 20]
```diff
@@ -0,0 +1,884 @@
...
+
+增量快照读取提供了在区块级别执行检查点的能力。它使用旧的快照读取机制解决了以前版本中的检查点超时问题。
+
+#### 无锁
+
+The MySQL CDC source 使用 **增量快照算法**, 避免获取全局读锁（使用读锁刷新表），因此不需要 “RELOAD” 权限。
```
Comment (leonardBang) [inline docs/content/connectors/mysql-cdc(ZH).md:375]: MySQL CDC source 使用 **增量快照算法**, 避免了数据库锁的使用，因此不需要 “RELOAD” 权限。

File: docs/content/connectors/mysql-cdc(ZH).md [hunk 21]
```diff
@@ -0,0 +1,884 @@
...
+
+ ```
+ (-∞, 'uuid-001'),
+ ['uuid-001', 'uuid-009'),
+ ['uuid-009', 'uuid-abc'),
+ ['uuid-abc', 'uuid-def'),
+ [uuid-def, +∞).
+```
+
+##### Chunk Reading Algorithm
```
Comment (leonardBang) [inline docs/content/connectors/mysql-cdc(ZH).md:447]: Chunk 读取算法

File: docs/content/connectors/mysql-cdc(ZH).md [hunk 22]
```diff
@@ -0,0 +1,884 @@
...
+* (3) 将当前 binlog 位置记录为`HIGH`偏移量
+* (4) 从`LOW`偏移量到`HIGH`偏移量读取属于快照区块的 binlog 记录
+* (5) 将读取的 binlog 记录向上插入缓冲区块记录，并发出缓冲区中的所有记录作为快照区块的最终输出（全部作为插入记录）
+* (6) 继续读取并发出属于 *单个 binlog reader* 中`HIGH`偏移量之后的区块的 binlog 记录。
+
+该算法的灵感来自 [DBLog Paper](https://arxiv.org/pdf/2010.12597v1.pdf), 请参考它了解更多详细信息。
+
+**Note:** 如果主键的实际值在其范围内分布不均匀，则在增量快照读取时可能会导致任务不平衡。
+
+### Exactly-Once Processing
```
Comment (leonardBang) [inline docs/content/connectors/mysql-cdc(ZH).md:463]:
```suggestion
### Exactly-Once 处理
```

File: docs/content/connectors/mysql-cdc(ZH).md [hunk 23]
```diff
@@ -0,0 +1,884 @@
...
+
+MySQL CDC 连接器是一个 Flink Source 连接器，它将首先读取表快照块，然后继续读取 binlog，
+无论是在快照阶段还是读取 binlog 阶段，MySQL CDC 连接器都会在处理时**准确读取数据**，即使任务出现了故障。
+
+### Startup Reading Position
```
Comment (leonardBang) [inline docs/content/connectors/mysql-cdc(ZH).md:468]: 启动模式

File: docs/content/connectors/mysql-cdc(ZH).md [hunk 24]
```diff
@@ -0,0 +1,884 @@
...
+      // set 4 parallel source tasks
+      .setParallelism(4)
+      .print().setParallelism(1); // 对 sink 设置并行度 1 以保持消息产出顺序
+
+    env.execute("Print MySQL Snapshot + Binlog");
+  }
+}
+```
+
+**Note:** 请参考 [Deserialization](../about.html#deserialization) 有关 JSON 反序列化的更多详细信息。
```
Comment (leonardBang) [inline docs/content/connectors/mysql-cdc(ZH).md:513]:
```suggestion
**注意:** 请参考 [Deserialization](../about.html#deserialization) 有关 JSON 反序列化的更多详细信息。
```

File: docs/content/connectors/mysql-cdc(ZH).md [hunk 25]
```diff
@@ -0,0 +1,884 @@
...
+
+### 扫描新添加的表
```
Comment (leonardBang) [inline docs/content/connectors/mysql-cdc(ZH).md:515]: 扫描新添加的表 --》 动态加表

File: docs/content/connectors/mysql-cdc(ZH).md [hunk 26]
```diff
@@ -0,0 +1,884 @@
...
+_Step 3_: 从 savepoint 还原更新后的 Flink 作业。
+```shell
+$ ./bin/flink run \
+      --detached \ 
+      --fromSavepoint /tmp/flink-savepoints/savepoint-cca7bc-bb1e257f0dab \
+      ./FlinkCDCExample.jar
+```
+**Note:** 请参考文档 [Restore the job from previous savepoint](https://nightlies.apache.org/flink/flink-docs-release-1.14/docs/deployment/cli/#command-line-interface) 了解更多详细信息。
+
+Data Type Mapping
```
Comment (leonardBang) [inline docs/content/connectors/mysql-cdc(ZH).md:572]: 数据类型映射

**Review by leonardBang — COMMENTED** [review]
Thanks @rookiegao for the great work, I left some comments.

**Comment by rookiegao** [comment]
I will modify it according to the suggestions. Thank you for your review.

**Comment by rookiegao** [comment]
The modification has been completed, please review again.

**Review by leonardBang — COMMENTED** [review]
Note: please use `rebase master` instead of `merge master` when update your PR branch

**Comment by rookiegao** [comment]
> Note: please use `rebase master` instead of `merge master` when update your PR branch

Thanks for the advice.

File: docs/content/connectors/mysql-cdc(ZH).md [hunk 27]
```diff
@@ -0,0 +1,883 @@
...
+|-----------------------------------------------------------|--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|-------------------------|
+| [mysql-cdc](mysql-cdc.md)         | <li> [MySQL](https://dev.mysql.com/doc): 5.6, 5.7, 8.0.x <li> [RDS MySQL](https://www.aliyun.com/product/rds/mysql): 5.6, 5.7, 8.0.x <li> [PolarDB MySQL](https://www.aliyun.com/product/polardb): 5.6, 5.7, 8.0.x <li> [Aurora MySQL](https://aws.amazon.com/cn/rds/aurora): 5.6, 5.7, 8.0.x <li> [MariaDB](https://mariadb.org): 10.x <li> [PolarDB X](https://github.com/ApsaraDB/galaxysql): 2.0.1 | JDBC Driver: 8.0.21     |
```
Comment (leonardBang) [inline docs/content/connectors/mysql-cdc(ZH).md:9]:
```suggestion
| [mysql-cdc](mysql-cdc(ZH).md)         | <li> [MySQL](https://dev.mysql.com/doc): 5.6, 5.7, 8.0.x <li> [RDS MySQL](https://www.aliyun.com/product/rds/mysql): 5.6, 5.7, 8.0.x <li> [PolarDB MySQL](https://www.aliyun.com/product/polardb): 5.6, 5.7, 8.0.x <li> [Aurora MySQL](https://aws.amazon.com/cn/rds/aurora): 5.6, 5.7, 8.0.x <li> [MariaDB](https://mariadb.org): 10.x <li> [PolarDB X](https://github.com/ApsaraDB/galaxysql): 2.0.1 | JDBC Driver: 8.0.21     |
```

File: docs/content/connectors/mysql-cdc(ZH).md [hunk 28]
```diff
@@ -0,0 +1,883 @@
...
+
+Dependencies
```
Comment (leonardBang) [inline docs/content/connectors/mysql-cdc(ZH).md:11]: 依赖

File: docs/content/connectors/mysql-cdc(ZH).md [hunk 29]
```diff
@@ -0,0 +1,883 @@
...
+    </tr>
+    <tr>
+      <td>
+        TINYBLOB<br>
+        BLOB<br>
+        MEDIUMBLOB<br>
+        LONGBLOB<br>
+      </td>
+      <td>BYTES</td>
+      <td>Currently, for BLOB data type in MySQL, only the blob whose length isn't greater than 2,147,483,647(2 ** 31 - 1) is supported. </td>
```
Comment (leonardBang) [inline docs/content/connectors/mysql-cdc(ZH).md:764]: 这里漏拉

File: docs/content/connectors/mysql-cdc(ZH).md [hunk 30]
```diff
@@ -0,0 +1,883 @@
...
+       MULTILINESTRING<br>
+       MULTIPOLYGON<br>
+       GEOMETRYCOLLECTION<br>
+      </td>
+      <td>
+        STRING
+      </td>
+      <td>
+      The spatial data types in MySQL will be converted into STRING with a fixed Json format.
+      Please see <a href="#空间数据类型映射 ">空间数据类型映射</a> section for more detailed information.
```
Comment (leonardBang) [inline docs/content/connectors/mysql-cdc(ZH).md:812]: 这几个类型也翻译下吧

File: docs/content/connectors/mysql-cdc(ZH).md [hunk 31]
```diff
@@ -0,0 +1,883 @@
...
+The field `coordinates` represents the `coordinates` of the spatial data.
+
+For `GEOMETRYCOLLECTION`, it will be converted into Json String with a fixed format like:<br>
+```json
+{"srid": 0 , "type": "GeometryCollection", "geometries": [{"type":"Point","coordinates":[10,10]}]}
+```
+
+The field `geometries` is an array contains all spatial data.
+
+The example for different spatial data types mapping is as follows:
```
Comment (leonardBang) [inline docs/content/connectors/mysql-cdc(ZH).md:838]: 这段可以翻译下

**Comment by rookiegao** [comment]
It is in rebase. After completion, it will be modified according to the suggestions

**Comment by rookiegao** [comment]
I made a mess... I want to open a PR again

**Comment by rookiegao** [comment]
This PR content has been transferred to #1435

**Pull request closed without merging**